bigquery/bigquerystorage/src/main/java/com/example/bigquerystorage/StorageArrowSample.java

```diff
@@ -0,0 +1,192 @@
+          new ByteArrayReadableSeekableByteChannel(
+              arrowSchema.getSerializedSchema().toByteArray())));
+      Preconditions.checkNotNull(schema);
+      List<FieldVector> vectors = new ArrayList<>();
+      for (Field field : schema.getFields()) {
+        vectors.add(field.createVector(allocator));
+      }
+      root = new VectorSchemaRoot(vectors);
+      loader = new VectorLoader(root);
+    }
+
+    /**
+     * Sample method for processing Arrow data which only validates decoding.
+     *
+     * @param batch object returned from the ReadRowsResponse.
+     */
+    public void processRows(ArrowRecordBatch batch) throws IOException {
+      org.apache.arrow.vector.ipc.message.ArrowRecordBatch deserializedBatch = MessageSerializer
+          .deserializeRecordBatch(
+              new ReadChannel(new ByteArrayReadableSeekableByteChannel(
+                  batch.getSerializedRecordBatch().toByteArray())),
+              allocator);
+
+      loader.load(deserializedBatch);
+      // Release buffers from batch (they are still held in the vectors in root).
+      deserializedBatch.close();
+      System.out.println(root.contentToTSVString());
+      // Release buffers from vectors in root.
+      root.clear();
+
+    }
+
+    @Override
+    public void close() {
+      root.close();
+      allocator.close();
+    }
+  }
+
+  public static void main(String... args) throws Exception {
+    // Sets your Google Cloud Platform project ID.
+    // String projectId = "YOUR_PROJECT_ID";
+    String projectId = args[0];
+    Integer snapshotMillis = null;
+    if (args.length > 1) {
+      snapshotMillis = Integer.parseInt(args[1]);
+    }
+
+    try (BigQueryStorageClient client = BigQueryStorageClient.create()) {
+      String parent = String.format("projects/%s", projectId);
+
+      // This example uses baby name data from the public datasets.
+      TableReference tableReference =
+          TableReference.newBuilder()
+              .setProjectId("bigquery-public-data")
+              .setDatasetId("usa_names")
+              .setTableId("usa_1910_current")
+              .build();
+
+      // We specify the columns to be projected by adding them to the selected fields,
+      // and set a simple filter to restrict which rows are transmitted.
+      TableReadOptions options =
+          TableReadOptions.newBuilder()
+              .addSelectedFields("name")
+              .addSelectedFields("number")
+              .addSelectedFields("state")
+              // Arrow doesn't support filters yet.
+              //.setRowRestriction("state = \"WA\"")
+              .build();
+
+      // Begin building the session request.
+      CreateReadSessionRequest.Builder builder =
+          CreateReadSessionRequest.newBuilder()
+              .setParent(parent)
+              .setTableReference(tableReference)
+              .setReadOptions(options)
+              // This API can also deliver data serialized in Apache Arrow format.
```

Comment:
s/Arrow/Avro/ since this is the inverted sample.

Reply:
fixed.